### ISSUE TITLE
enable cuda8.0 for taichi request!

### ISSUE THREAD

**Issue by jakoaltrd** [taichi-dev/taichi#2296]
hi, everyone, I 'd like to use taichi with cuda8.0 on my old laptop, and I think it will benefit some others, so I 
post this request.

to achieve that, first, I need to check if taichi(version) support cuda8.0, so how to check this?
second, if cuda8.0 are not supported, then I need to build taichi from the source,then how?

plz leave your answers and advieces here, and I will post the results and final solution for this.

**Comment by k-ye** [comment]
Hi @jakoaltrd ,

Please refer to https://taichi.readthedocs.io/en/latest/dev_install.html for how to build taichi from source. In addition, could you try installing OpenGL 4.3 and use the `ti.opengl` backend? Thanks!

**Comment by jakoaltrd** [comment]
@k-ye 
I checked my OpenGL version and it is 4.6, ti.opengl works, but it is way too slow, how to set parameters in taichi to enable multi cores or multi threads to boost the performance?

**Comment by jakoaltrd** [comment]
@k-ye hi, I tried my opengl4.6 backend, it is working but I had a question:
    when I init taichi with ti.arch = opengl, does it use opengl's compute shader
as backend to do calculation?

**Comment by bobcao3** [comment]
> 
> 
> @k-ye hi, I tried my opengl4.6 backend, it is working but I had a question:
> when I init taichi with ti.arch = opengl, does it use opengl's compute shader
> as backend to do calculation?

Yes it does. On my end OpenGL is not drastically slower than CUDA, there's some difference but I'd say less than 20%, probably more along the lines of 10%
